### PR TITLE
Concurrent execution of Test262Test

### PR DESCRIPTION
**Pull request by doehyunbaek**
Support concurrent execution of Test262Test, this is mainly for performance reason.

For the Linux desktop with 12900F, this reduces test time of whole suite from  01:22:22 to 16:41, the benefit may vary between the systems.

Support serial extraction of ecma262 specification. This is mainly to help debugging as debugging concurrent execution of Extractor was a little bit confusing.

### CODE REVIEW

**Comment by doehyunbaek** [comment]
I will fix the compl test today.

File: src/main/scala/esmeta/extractor/Extractor.scala [hunk 1]
```diff
@@ -18,10 +18,15 @@ object Extractor:
     document: Document,
     version: Option[Spec.Version],
     eval: Boolean,
-  ): Spec = new Extractor(document, version, eval).result
+    concurrent: Boolean,
+  ): Spec = new Extractor(document, version, eval, concurrent).result
```
Comment (jhnaldo) [inline src/main/scala/esmeta/extractor/Extractor.scala]: Is it necessary to support concurrent extraction for concurrent Test262 testing?

File: src/main/scala/esmeta/phase/Test262Test.scala [hunk 2]
```diff
@@ -26,6 +26,8 @@ case object Test262Test extends Phase[CFG, Summary] {
   ): Summary =
     // set test mode
     TEST_MODE = true
+    if (config.concurrent && config.progress)
+      warn("progress bar may be unstable in concurrent mode.")
```
Comment (jhnaldo) [inline src/main/scala/esmeta/phase/Test262Test.scala]: Is it impossible to resolve this issue?

File: src/main/scala/esmeta/util/ProgressBar.scala [hunk 3]
```diff
@@ -78,17 +79,26 @@ case class ProgressBar[T](
...
       show
-
-    for ((x, idx) <- iterable.zipWithIndex)
-      val name = getName(x, baseSize + idx)
-      getError {
-        f(x)
-        summary.pass.add(name)
-      }.map(errorHandler(_, summary, name))
-      gcount += 1
+    if (concurrent)
```
Comment (jhnaldo) [inline src/main/scala/esmeta/util/ProgressBar.scala]: Is it possible to design a more unified form?

**Review by jhnaldo — CHANGES_REQUESTED** [review]

File: src/main/scala/esmeta/extractor/Extractor.scala (hunk 1 above)
Comment (doehyunbaek) [reply src/main/scala/esmeta/extractor/Extractor.scala]: Not necessary. Excluded from this PR.

File: src/main/scala/esmeta/phase/Test262Test.scala (hunk 2 above)
Comment (doehyunbaek) [reply src/main/scala/esmeta/phase/Test262Test.scala]: Resolved with https://github.com/es-meta/esmeta/pull/171/files#diff-eaae679fb425688c5603f0abb743dd3c8150770095868ba59a4b913486406b04R34

File: src/main/scala/esmeta/util/ProgressBar.scala (hunk 3 above)
Comment (doehyunbaek) [reply src/main/scala/esmeta/util/ProgressBar.scala]: Tried with https://github.com/es-meta/esmeta/pull/171/files#diff-f61fd7e6c4e7d373cb7247e2a26c4354361fc73c96f4e9189176365bc6cd30f5R82-R89 .

**Comment by doehyunbaek** [comment]
I observed the execution of concurrent execution diverging from the serial one. Will open again after I investigate the cause.

**Comment by doehyunbaek** [comment]
> I observed the execution of concurrent execution diverging from the serial one. Will open again after I investigate the cause.

Sorry, there was no discrepancy. There was a change from number of passing tests from [this commit](https://github.com/es-meta/esmeta/actions/runs/5619109352/job/15225697923) to [this commit](https://github.com/es-meta/esmeta/actions/runs/5758340815/job/15610737207), which made a confusion.

I confirmed the concurrent and serial version outputting the exact same summary

**Review by jhnaldo — APPROVED** [review]
I slightly refactored this PR and supported the progress bar with the concurrent `Summary`. After passing the CI system, I will merge it.

**Pull request merged**